pkg/gofr/datasource/pubsub/nats/config.go

```diff
@@ -0,0 +1,76 @@
+package nats
+
+import (
+	"time"
+
+	"gofr.dev/pkg/gofr/datasource/pubsub"
+)
+
+// Config defines the Client configuration.
+type Config struct {
+	Server      string
+	CredsFile   string
+	Stream      StreamConfig
+	Consumer    string
+	MaxWait     time.Duration
+	MaxPullWait int
+	BatchSize   int
+}
+
+// StreamConfig holds stream settings for NATS JetStream.
+type StreamConfig struct {
+	Stream     string
+	Subjects   []string
+	MaxDeliver int
+	MaxWait    time.Duration
+	MaxBytes   int64
+}
+
+// New creates a new Client.
+func New(cfg *Config, logger pubsub.Logger) *PubSubWrapper {
+	if cfg == nil {
+		cfg = &Config{}
+	}
+
+	if cfg.BatchSize == 0 {
+		cfg.BatchSize = 100 // Default batch size
```

Comment:
Let's make default batchSize a constant , in this way even if we want to change it someday it will be a single line change.

Reply:
updated